app/code/community/Algolia/Algoliasearch/Helper/Entity/Producthelper.php

```diff
@@ -512,102 +521,100 @@ protected function handlePrice(Mage_Catalog_Model_Product &$product, $sub_produc
-                $customData[$field][$currency_code]['special_to_date'] = strtotime($product->getSpecialToDate());
-
-                if ($customer_groups_enabled) {
-                    foreach ($groups as $group) {
-                        $group_id = (int) $group->getData('customer_group_id');
-
-                        if ($special_price && $special_price < $customData[$field][$currency_code]['group_'.$group_id]) {
-                            $customData[$field][$currency_code]['group_'.$group_id.'_original_formated'] =
-                                $customData[$field][$currency_code]['default_formated'];
-
-                            $customData[$field][$currency_code]['group_'.$group_id] = $special_price;
-                            $customData[$field][$currency_code]['group_'.$group_id.'_formated'] = $this->formatPrice(
-                                $special_price,
+                        if ($canSpecialBeValid && $special_price < $cdfc[$group_str]) {
+                            $cdfc[$group_str .'_original_formated'] = $this->formatPrice(
+                                $cdfc[$group_str],
                                 false,
                                 $currency_code
                             );
+                            $cdfc[$group_str .'_original'] = $cdfc[$group_str];
+
+                            $cdfc[$group_str] = $special_price;
                         }
+                        $cdfc[$group_str .'_formated'] = $store->formatPrice($cdfc[$group_str], false, $currency_code);
                     }
+                    $product->setCustomerGroupId(null);
                 }
 
-                if ($special_price && $special_price < $customData[$field][$currency_code]['default']) {
-                    $customData[$field][$currency_code]['default_original_formated'] =
-                        $customData[$field][$currency_code]['default_formated'];
+                if ($canSpecialBeValid && $special_price < $cdfc['default']) {
+                    $cdfc['default_original_formated'] = $cdfc['default_formated'];
+                    $cdfc['default_original'] = $cdfc['default'];
 
-                    $customData[$field][$currency_code]['default'] = $special_price;
-                    $customData[$field][$currency_code]['default_formated'] = $this->formatPrice(
+                    $cdfc['default'] = $special_price;
+                    $cdfc['default_formated'] = $this->formatPrice(
                         $special_price,
                         false,
                         $currency_code
                     );
                 }
 
+                // configurable appears to assume that there are no customer_group_prices
+                // specified on the attached simple products, but needs re-writing anyway
+                // to correctly calculate the option price, not the simple price
                 if ($type == 'grouped' || $type == 'bundle' || $type == 'configurable') {
```

Comment:
From here, these changes are just to use the $cdfc variable, with the exception of the one comment I've added